src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts

```diff
@@ -29,12 +28,9 @@ export const useGetLimitOrderEvents = () => {
     [value],
   );
 
-  const { events } = useGetContractPastEvents(
-    'settlement',
-    'MarginOrderFilled',
-  );
+  const { data } = useGetMarginLimitOrderFilled();
 
-  useLog('LimitOrderTables', events);
+  const events = useMemo(() => data?.marginOrderFilleds, [data]);
```

Comment:
Pls ask Betsy/Remy to correct the typo, should be `marginOrderFilled`.

Reply:
Hi guys, the singular/plural naming is a graphql thing. If you want multiple MarginOrderFilled entities, they are called marginOrderFilled and you have to provide an ID. If you want multiple entities, they are called marginOrderFilleds

Reply:
Thanks Betsy! @tiltom is it ok for you?

Reply:
well yeah, it looks a bit weird but I understand the explanation :D

Reply:
I hate this too but appreciate the fact you can't change it 😂